src/reducers/versions.tsx

```diff
@@ -197,21 +208,37 @@ export const getVersionFiles = (
   return versions.versionFiles[versionId];
 };
 
-export const getVersionFile = (
+export const getVersionInfo = (
   versions: VersionsState,
   versionId: VersionId,
-  path: string,
 ) => {
-  const filesForVersion = getVersionFiles(versions, versionId);
-
-  return filesForVersion ? filesForVersion[path] : undefined;
+  return versions.versionInfo[versionId];
 };
 
-export const getVersionInfo = (
+export const getVersionFile = (
   versions: VersionsState,
   versionId: VersionId,
-) => {
-  return versions.versionInfo[versionId];
+  path: string,
+): VersionFile | void => {
+  const version = getVersionInfo(versions, versionId);
+  const filesForVersion = getVersionFiles(versions, versionId);
+
+  if (version && filesForVersion) {
+    const file = filesForVersion[path];
+    const entry = versions.versionInfo[versionId].entries.find(
+      (e) => e.path === path,
+    );
+    if (entry && file) {
+      return {
+        ...file,
+        mimeType: entry.mimeType,
+        type: entry.type,
+      };
+    }
+  }
+
+  // The version or an entry was not found.
+  return undefined;
```

Comment:
If `entry` is missing, we should log a debug message because that'd be abnormal.

Reply:
Hmm, `getVersionFile` is a selector, and we usually don't give them objects as arguments, but if I want to add a `_log` argument in order to test the logging we should probably change the signature to accept an object.

I could either just use `log` in the selector and not add a test for it, or I could change the signature to allow us to pass a `_log` argument into `getVersionFile` in order to test the log part.

Which do you think is better, @willdurand?

Reply:
I think you can simply add a 4th argument:

```js
export const getVersionFile = (
  versions: VersionsState,
  versionId: VersionId,
  path: string,
  _log = log,
): VersionFile | void => {
  // ...
};
```

Reply:
I prefer to see options as objects because it's easier to grow the function later:

```js
export const getVersionFile = (
  versions: VersionsState,
  versionId: VersionId,
  path: string,
  { _log = log } = {},
): VersionFile | void => {
  // ...
};
```

Reply:
Thanks @willdurand and @kumar303. The options object is a good idea. I have updated the patch to do that.